src/apps/HeyGPT/HeyGPT/Helpers/DebuggerHelper.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HeyGPT.App.Helpers;
+public static class DebuggerHelper
+{
+    /// <summary>
+    /// Waits for a debugger to attach by showing a message box.
+    /// This method is conditional on WAIT_FOR_DEBUGGER_TO_ATTACH and only executes in DEBUG builds.
+    /// </summary>
+    [Conditional("WAIT_FOR_DEBUGGER_TO_ATTACH")]
+    public static void WaitForDebuggerToAttach()
```

Comment:
When trying to debug protocols and activation events you sometimes need to attach the debugger instead of running in the IDE.   This mechanism is an old trick to stop the program while you can attach the debugger.  Close the message window and you continue the program and hit your early breakpoint.